app/models/topic.rb

```diff
@@ -1802,7 +1802,12 @@ def rate_limit_topic_invitation(invited_by)
   end
 
   def cannot_permanently_delete_reason(user)
-    if self.posts_count > 0
+    deleted_posts_count = Post.with_deleted
```

Comment:
Are we missing a `.where.not(deleted_at: nil)` here? Do you think it's worth it to extract this into a scope?

Reply:
That's exactly what the `with_deleted` scope does 😉

Reply:
Actually, @romanrizzi is right, `Post.with_deleted` will be giving all posts (including those that are deleted), not just the ones that are deleted as the name of the variable suggests (`deleted_posts_count`).

Reply:
Maybe just rename the variable to `with_deleted_posts_count`?

Reply:
I renamed it to `all_posts_count`. :+1: